components/gateway/internal/auditlog/client.go

```diff
@@ -76,22 +76,37 @@ func (c *Client) LogSecurityEvent(ctx context.Context, event model.SecurityEvent
 
 func (c *Client) sendAuditLog(ctx context.Context, req *http.Request) error {
 	logger := log.C(ctx)
+	responseStatusCode, err := c.sendAuditLogInternal(ctx, req)
+	if responseStatusCode != http.StatusCreated || err != nil {
```

Comment:
Consider using github.com/avast/retry-go in order to mitigate different http req / body issues.

I think this is the library that is used across the project, you can find some usages for reference.